Library/Homebrew/formulary.rb

```diff
@@ -112,10 +112,10 @@ def initialize(bottle_name)
         resource = Resource.new(formula_name) { url bottle_name }
         resource.specs[:bottle] = true
         downloader = CurlDownloadStrategy.new resource.name, resource
-        @bottle_filename = downloader.cached_location
-        cached = @bottle_filename.exist?
+        cached = downloader.cached_location.exist?
         downloader.fetch
         ohai "Pouring the cached bottle" if cached
+        @bottle_filename = downloader.cached_location
```

Comment:
What's the reordering about; does this change?

Reply:
This doesn't change anything at the moment, but this ordering is necessary for what I have in mind for refactoring the download locations.

Instead of relying on a somewhat hacky way to get the proper extension from a URL *before* downloading, we use no extension when initializing the download. We then know the actual file name *after* the download is finished, so in this case `@bottle_filename` should be set only after `downloader.fetch` has been called.

Reply:
So currently a cancelled download would have a correct extension – whether guessed in a "hacky" manner or not – and after the change a cancelled download would be extensionless? That sounds like a regression to me.

Reply:
What use does a correct extension have when the download is cancelled?

Reply:
Most of the downloads can be resumed – please _do not_ remove that ability as it's very useful – but `foo-1.2.3.tar.bz2.incomplete` shouldn't be resumable if the extension changed from bz2 to xz. Also it's much more clear what a `foo-1.2.3.tar.bz2.incomplete` is than what a `foo-1.2.3.incomplete` is.